libexec/job/slurm/parser.sh

```diff
@@ -27,9 +27,20 @@
 #==============================================================================
 
 # ==============================================================================
-# State Mapping and helper functions
+# Parsing functions and helpers specific to Slurm but not specific to any
+# particular Slurm CLI, e.g., `scontrol` or `sacct.
 # ==============================================================================
 
+source_parsers() {
```

Comment:
A script could use both `scontrol` and  `sacct`.  It just needs to (re)load the correct parsers before parsing the result.  If bash had a better namespace mechanism / better support for polymorphism I'd use it.  Unfortunately this is as good as I know how to get it.

The current simple implementation allows a script to do the following.  I don't see that we want to prevent that.

```
out=$(scontrol ...)
source_parsers "scontrol"
parse_stuff

out=$(sacct ...)
source_parsers "sacct"
parse_stuff

out=$(scontrol ...)
source_parsers "scontrol"
parse_more_stuff
```